### ISSUE TITLE
Filter network connections with bootnodes

### ISSUE THREAD

**Issue by Firescar96** [ethereum/go-ethereum#16057]
#### Expected behaviour
I set up a PoA network with three nodes on AWS, all publicly accessible. A fourth node is running using the lightweight `bootnode` client. The bootnode should rapidly facilitate connections between these nodes.

#### Actual behaviour
There is no ability to specify a network id so when I tried to add another node after a few days via the bootnode I couldn't. The bootnode becomes saturated with nodes on other networks and is unable to do matchmaking for me anymore

#### Discussion
Full nodes do not form persistent connections with nodes on other networks. When I run `admin.peers` from the geth console of a full node it usually only shows peers with the same network id. Other potential peers are ephemeral. How can I help to provide the same behavior to bootnodes, or is it recommended that I just use a full node as a bootnode?

**Comment by karalabe** [comment]
Hi there!

Yes, we are aware of this limitation. Unfortunately the currently stable discovery protocol within Ethereum does not facilitate filtering for specific capabilities of nodes, and as such everyone gets conflated into one big mess.

For quite a while now the light client in Geth has been shipping with a next generation experimental discovery protocol that **can** do filtering and lookups based on certain user criteria. Light servers, light clients and light bootnodes already use that. Unfortunately it's not yet been stabilized and standardized to the point where we'd be confident to switch over the entire network to it. It's sadly a fairly low priority task among all the other juicy scaling issues.

On the upside, we've been working on standardizing Ethereum Node Records (https://github.com/ethereum/EIPs/pull/778), which would define a way for any Ethereum node to list its capabilities; and also to allow remotely retrieving those (https://github.com/ethereum/EIPs/pull/868). These two alone are not fully enough to fix your use case, but these would allow a third extension not yet published to add new discovery mechanisms (e.g. DNS based ones), which would serve as both a fallback as well as alternative way to find peers.

TL;DR We don't have a better solution for now, but we're exploring and implementing multiple concurrent solutions.

**Comment by marcosmartinez7** [comment]
Hi, is there any progress with this about filtering the interactions between the bootnodes?

Kind regards
Marcos
